torch/optim/sparse_adam.py

```diff
@@ -97,11 +97,36 @@ def step(self, closure=None):
 
         return loss
 
-SparseAdam.__doc__ = r"""Implements lazy version of Adam algorithm suitable for sparse tensors.
-
-    In this variant, only moments that show up in the gradient get updated, and
-    only those portions of the gradient get applied to the parameters.
-
+SparseAdam.__doc__ = r"""SparseAdam implements a lazy version of the Adam algorithm
+    suitable for sparse gradients. Please note that SparseAdam is NOT equivalent to
+    Adam "but for sparse gradients", and we understand this may be confusing.
+
+    This is a semantically different optimizer where only moments that show up in the
+    gradient get updated, and only those portions of the gradient get applied to the
+    parameters. This is an optimization, since the 0-values in the gradient will be
+    skipped, and one can presume that this is only an optimization in semantically
+    sparse tensors, meaning there are lots of 0-values that are skipped. We implement
```

Comment:
> This is an optimization, since the 0-values in the gradient will be skipped, and one can presume that this is only an optimization in semantically **sparse tensors**, meaning there are lots of 0-values that are skipped.

Maybe would help to clarify, is it sparse input tensors that this would apply for?

Reply:
No, I mean gradients here too. I'll clarify!